packages/botonic-plugin-contentful/src/tools/l10n/import-updater.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  BotonicContentType,
+  ContentId,
+  ContentType,
+  MessageContentType,
+  TopContentId,
+} from '../../cms'
+import { ContentFieldType, ManageCms, ManageContext } from '../../manage-cms'
+import { ContentDeleter } from '../../manage-cms/content-deleter'
+import {
+  CONTENT_FIELDS,
+  getFieldsForContentType,
+} from '../../manage-cms/fields'
+import { isOfType } from '../../util/enums'
+import { Record, RecordFixer, recordId } from './csv-import'
+
+export class ContentToImport {
+  constructor(
+    readonly contentId: ContentId,
+    readonly name: string,
+    readonly fields: { [field: string]: any }
+  ) {}
+}
+
+/**
+ * Reduce all the records read from a csv into ContentToImport's
+ * and delegates to ContentUpdater.
+ * It also checks consistency of all records of a given content
+ */
+export class ImportRecordReducer {
+  pending: Record[] = []
+
+  constructor(readonly importer: ImportContentUpdater) {}
+
+  async consume(record: Record): Promise<void> {
+    if (!isOfType(record.Model, ContentType)) {
+      console.error(
+        `Bad model '${String(record.Model)}'. Should be one of ${String(
+          Object.values(BotonicContentType)
+        )}`
+      )
+      return
+    }
+
+    const field = CONTENT_FIELDS.get(record.Field)
+    if (!field) {
+      console.error(`Bad field '${record.Field}'`)
+      return
+    }
+    if (!this.checkLast(record)) {
+      return
+    }
+    const last = this.last()
+    if (last) {
+      if (last.Id != record.Id) {
+        await this.flush()
+      }
+    }
+    this.pending.push(record)
+  }
+
+  checkLast(record: Record): boolean {
+    const last = this.last()
+    if (!last) return true
+    let diffField: string | undefined = undefined
+
+    if (last.Model != record.Model) {
+      diffField = 'model'
+    }
+    if (last.Code != record.Code) {
+      diffField = 'code'
+    }
+    if (diffField) {
+      console.error(
+        `Records ${recordId(record)} & ${recordId(
+          last
+        )} with same id have different ${diffField}`
+      )
+      return false
+    }
```

Comment:
How do you know that `record` and `last` have the same Id? I can't see where are you checking it.

Reply:
you're a crack reviewer!
yes, it's fixed in next PR #1117 
Anyway I added a relevant unit test at 103b8e6